client/src/main/java/io/pravega/client/stream/impl/ReaderGroupState.java

```diff
@@ -90,6 +90,19 @@
     @GuardedBy("$lock")
     private final Map<SegmentWithRange, Long> lastReadPosition;
     private final Map<Segment, Long> endSegments;
+    @Getter
+    private final long generation;
+    @Getter
+    private final ConfigState configState;
+    @Getter
+    private final ReaderGroupConfig newConfig;
+
+    public enum ConfigState {
```

Comment:
These config states in essence define a state machine. Could you document via javadoc what these states mean and what the transitions of the state machine are?

Reply:
Documentation for the enum can be added to explain the states and the transitions involved.